hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java

```diff
@@ -437,4 +447,117 @@ private static ServiceException createLeaderNotReadyException(
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    case DeleteKey:
+      volName = omRequest.getDeleteKeyRequest().getKeyArgs().getVolumeName();
+      buckName = omRequest.getDeleteKeyRequest().getKeyArgs().getBucketName();
+      buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    case CreateFile:
+      volName = omRequest.getCreateFileRequest().getKeyArgs().getVolumeName();
+      buckName = omRequest.getCreateFileRequest().getKeyArgs().getBucketName();
+      buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    case CreateDirectory:
+      volName =
+          omRequest.getCreateDirectoryRequest().getKeyArgs().getVolumeName();
+      buckName =
+          omRequest.getCreateDirectoryRequest().getKeyArgs().getBucketName();
+      buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    case RenameKey:
+      volName = omRequest.getRenameKeyRequest().getKeyArgs().getVolumeName();
+      buckName = omRequest.getRenameKeyRequest().getKeyArgs().getBucketName();
+      buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    case InitiateMultiPartUpload:
+      volName = omRequest.getInitiateMultiPartUploadRequest().getKeyArgs()
+          .getVolumeName();
+      buckName = omRequest.getInitiateMultiPartUploadRequest().getKeyArgs()
+          .getBucketName();
+      buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    case CommitMultiPartUpload:
+      volName = omRequest.getCommitMultiPartUploadRequest().getKeyArgs()
+          .getVolumeName();
+      buckName = omRequest.getCommitMultiPartUploadRequest().getKeyArgs()
+          .getBucketName();
+      buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    case AbortMultiPartUpload:
+      volName = omRequest.getAbortMultiPartUploadRequest().getKeyArgs()
+          .getVolumeName();
+      buckName = omRequest.getAbortMultiPartUploadRequest().getKeyArgs()
+          .getBucketName();
+      buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    case CompleteMultiPartUpload:
+      volName = omRequest.getCompleteMultiPartUploadRequest().getKeyArgs()
+          .getVolumeName();
+      buckName = omRequest.getCompleteMultiPartUploadRequest().getKeyArgs()
+          .getBucketName();
+      buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
+      bucketLayout = getBucketLayout(buckInfo);
+      break;
+    default:
+      break;
+    }
```

Comment:
`bucketLayout.equals(BucketLayout.FILE_SYSTEM_OPTIMIZED)`, there is a findbug warn, please modify like below.

bucketLayout can be null, please do `BucketLayout.FILE_SYSTEM_OPTIMIZED.equals(bucketLayout)`.

Also, please add a TODO to handle the case of null buckInfo.

Reply:
Done